pkg/promql/engine.go

```diff
@@ -42,19 +42,27 @@ import (
 	"github.com/prometheus/prometheus/storage"
 	"github.com/prometheus/prometheus/util/stats"
 
+	pgmodel "github.com/timescale/promscale/pkg/pgmodel/model"
 	mq "github.com/timescale/promscale/pkg/pgmodel/querier"
 	"github.com/timescale/promscale/pkg/util"
 )
 
 // A Queryable handles queries against a storage.
 // Use it when you need to have access to all samples without chunk encoding abstraction e.g promQL.
 type Queryable interface {
-	// Querier returns a new Querier on the storage.
-	Querier(ctx context.Context, mint, maxt int64) (Querier, error)
+	// SamplesQuerier returns a new promql.Querier on the storage. It helps querying over samples
+	// in the database.
+	Samples(ctx context.Context, mint, maxt int64) (SamplesQuerier, error)
+	// ExemplarQuerier returns a new Querier that helps querying exemplars in the database.
+	Exemplar(ctx context.Context) ExemplarQuerier
 }
 
-// Querier provides querying access over time series data of a fixed time range.
-type Querier interface {
+type ExemplarQuerier interface {
```

Comment:
Is this our thing? ever used? or is it an upstream change?

Reply:
our thing

Reply:
can we use the querier.ExemplarQuerier instead?